dataRetriever/factory/resolverscontainer/baseResolversContainerFactory.go

```diff
@@ -417,3 +417,48 @@ func (brcf *baseResolversContainerFactory) generateValidatorInfoResolver() error
 
 	return brcf.container.Add(identifierValidatorInfo, validatorInfoResolver)
 }
+
+func (brcf *baseResolversContainerFactory) generateReceiptResolver() error {
+	shardC := brcf.shardCoordinator
+
+	// only one shard receipt topic
+	identifier := factory.ReceiptTopic
```

Comment:
```suggestion
	identifier := factory.ReceiptTopic + shardC.CommunicationIdentifier(shardC.SelfId())
```
in order to make it intrashard

Reply:
done